extensions/wikia/WikiFactory/SpecialWikiFactory_body.php

```diff
@@ -1,56 +1,60 @@
 <?php
 
-/**
- * @package MediaWiki
- * @subpackage SpecialPage
- * @author Krzysztof Krzyżaniak <eloy@wikia-inc.com> for Wikia.com
- * @version: 0.1
- */
-
 if ( !defined( 'MEDIAWIKI' ) ) {
 	echo "This is MediaWiki extension and cannot be used standalone.\n";
 	exit( 1 ) ;
 }
 
-
 /**
  * @addtogroup SpecialPage
  */
 class WikiFactoryPage extends SpecialPage {
+	// Possible paths for the metrics tab
+	const METRIC_PATHS = [ "metrics", "metrics/main", "metrics/monthly", "metrics/daily" ];
+
+	public $mStatuses = [ -2 => 'spam', -1 => 'disabled*', "disabled", "enabled", "redirected" ];
+
+	/** @var WikiFactoryTags */
+	private $mTags;
 
-	private $mWiki, $mTitle, $mDomain, $mTab, $mVariableName, $mTags, $mSearchTag;
-	public $mStatuses = array(-2 => 'spam', -1=> 'disabled*', "disabled", "enabled", "redirected" );
-	private $mTagWikiIds = array();
+	private $mWiki, $mTitle, $mDomain, $mTab, $mVariableName, $mSearchTag;
+	private $mTagWikiIds = [];
+
+	private $mRemoveTag;
+	private $mRemoveTags;
+
+	private $mMassTag;
+	private $mMassTagWikis;
 
 	/**
 	 * constructor
 	 */
 	function  __construct() {
-		parent::__construct( "WikiFactory"  /*class*/, 'wikifactory' /*restriction*/);
+		parent::__construct( "WikiFactory" /*class*/, 'wikifactory' /*restriction*/ );
 	}
 
 	/**
 	 * execute
 	 *
 	 * main entry point
 	 *
-	 * @access public
-	 *
-	 * @param string $subpage: subpage of Title
+	 * @param string $subpage : subpage of Title
 	 *
-	 * @return nothing
+	 * @throws PermissionsError
+	 * @throws ReadOnlyError
+	 * @throws UserBlockedError
 	 */
 	public function execute( $subpage ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)